src/sonic-yang-models/yang-models/sonic-buffer-profile.yang

```diff
@@ -0,0 +1,67 @@
+module sonic-buffer-profile {
+	namespace "http://github.com/Azure/sonic-buffer-profile";
+	prefix bpf;
+
+	import sonic-buffer-pool {
+		prefix bpl;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC BUFFER PROFILE";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+
+	container sonic-buffer-profile {
+
+		container BUFFER_PROFILE {
+
+			list BUFFER_PROFILE_LIST {
+				key "name";
+
+				leaf name {
+					type string;
+				}
+
```

Comment:
As discussed in the meeting, can we add optional field mode (static/dynamic) as well?

Reply:
Venkat, there should not be a need to add an optional field "mode". Only one of static_th/dynamic_th must be specified and mode can be derived from threshold defined.  static_th/dynmic_th can be made mandatory and using choice  we can select one of then,

Reply:
This mode field is already present in the backend code and present in SAI spec as well under profile (SAI_BUFFER_PROFILE_ATTR_THRESHOLD_MODE), so, in order to match the existing implementation, we have to add it.

Reply:
Please check bufferorch.cpp, the mode value is derived based on static_th/dynamic_th value and 2 SAI Attributes are set accordingly (SAI_BUFFER_PROFILE_ATTR_THRESHOLD_MODE 
 and SAI_BUFFER_PROFILE_ATTR_SHARED_DYNAMIC_TH/SAI_BUFFER_PROFILE_ATTR_SHARED_STATIC_TH).

Reply:
I wanted to have the mode under buffer-profile table because some platforms use it, but since the backend is assuming the mode from static_th/dynamic_th, we can avoid the mode field in the profile table and update platform-specific buffer defaults, thanks!

https://github.com/Azure/sonic-buildimage/blob/master/device/dell/x86_64-dellemc_s5232f_c3538-r0/DellEMC-S5232f-C32/buffers_defaults_t0.j2#L26